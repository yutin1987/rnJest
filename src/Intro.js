import React, {Component} from 'react';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 30,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

export default class Intro extends Component {

  state = {
    value: '',
  }

  static defaultProps = {
  }

  onEnter = () => {
    console.log('mouse enter');
    this.setState({ value: 'XYZ111' });
  }

  render() {
    const { value } = this.state;

    console.log(value);

    return (
      <View style={styles.container} onEnter={this.onEnter}>
        <Text style={styles.welcome}>
          {`Welcome to React Native! ${value}`}
        </Text>
        <Text style={styles.instructions}>
          This is a React Native snapshot test.
        </Text>
      </View>
    );
  }
}
