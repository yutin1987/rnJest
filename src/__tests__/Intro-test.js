import 'react-native';
import React from 'react';
import Intro from '../Intro';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const component = renderer.create(
    <Intro />
  );

  const tree = component.toJSON();

  expect(component.toJSON()).toMatchSnapshot();

  tree.props.onEnter();

  expect(component.toJSON()).toMatchSnapshot();
});
