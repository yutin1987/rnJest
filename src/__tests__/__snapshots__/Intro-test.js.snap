exports[`test renders correctly 1`] = `
<View
  onEnter={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    Welcome to React Native! 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    This is a React Native snapshot test.
  </Text>
</View>
`;

exports[`test renders correctly 2`] = `
<View
  onEnter={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    Welcome to React Native! XYZ111
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    This is a React Native snapshot test.
  </Text>
</View>
`;
